### PR TITLE
Various Antagonist Handgun/Rifle changes; Stechkin, Stechkin APS, c20-r, m90gl

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Alters a few of the antagonist weapons, primarily the Soviet handguns.

**Stechkin**
Now a makarov, using the old Stechkin APS sprites with slight tweaks. It's now chambered for 9mm. Comes with the same ammo diversity as before. Same damage as before. As a consequence, all 9mm using guns do 30 damage. Wowzers.

**Stechkin APS**
Now actually a normal sized gun. This gun isn't small. It's about as big as a m1911 and apparently pretty heavy. It's now available in nuke op uplinks, mostly so you don't have to buy a specific bundle just to get it. Nukies could ALREADY buy this gun, it's just in a bundle. Now you can buy it separately. It also shoots fast and sprays bullets everywhere.

Screenshot 2020-05-04 10 29 13
Screenshot 2020-05-04 10 26 32

The Stechkin APS costs 10 TC. Same as the c20-r before. Because it literally does the same damage that the c20-r used to do. Shock horror. No ammo variety but nothing stopping you buying some 9mm ammo and filling the APS magazines with it.

**10mm Bullets**
Now do pretty sizable amounts of damage. Though the amount of guns that actually USE 10mm is...like, just the surplus rifle, flat guns I think and the old Stechkin. Really goddamn rare bullet type.

Basically does 10 more damage across the board.

**c20-R and most .45 guns**
.45 now does 35 damage. It's only slightly better per bullet than 9mm.

The c20-R now has more ammo options out of the uplink.

c20-R is now 13 TC. It shoots 3 bullets, that's kind of scary.

**m90gl and anything using 5.56mm**

Now has an innate 30 bullet penetration. This shouldn't appreciably change too much, as it was a one burst crit on an unarmored target before.

Added phasic 5.56mm. This ammunition passes through all objects, like penetrators do. It does significantly less damage per bullet. You can buy it for 8 TC.

## Why It's Good For The Game

It's honestly soothing to my OCD knowing that the soviet era guns are a little closer to their real world equivalents. The APS in particular was always a standout as a weird gun compared to it's real life version.

I kinda felt the c20-r was grossly overshadowed lately as far as a gun choices go. Hopefully this makes it more attractive.

## Changelog
:cl:
tweak: The stechkin pistol is now a makarov. It comes chambered in 9mm. Nothing has really changed for performance or ammo availability. It's just all in 9mm now.
tweak: The stechkin APS is now normal sized and available to nukies for 10 tc. It sprays bullets like mad and fires very quickly.
tweak: The caravan now has stechkin APS magazines.
tweak: The Spetznaz bundle uses incendiary magazines for the APS.
balance: .45 is stronger across the board by about 5 damage, and has more bullet variety.
balance: the c20-r is a 3 burst weapon and costs 13tc.
balance: 5.56mm now has 30 armor penetration and has some bullet spread.. This only affects the m90gl and the Boarder assault rifles.
tweak: You can buy phasic ammo for the m90 for 8 tc out of the uplink. It passes through walls and objects, and does significantly less damage. (comparatively speaking of course)
tweak: X-Ray beams now pass through doors, machinery, and structures like lockers.
/:cl:

### CODE REVIEW

**Comment by partyaddict** [comment]
we're in the year 2500, i believe the syndicate would use something more modern than a 600 year old soviet handgun

then again we have mosins and 1911s so i don't even know how i can make that judgement

**Comment by necromanceranne** [comment]
You do realize the stechkin is from the same era as the makarov, right?

**Comment by ATH1909** [comment]
Make sure that this won't break any map, away mission, or ruin designs.

For example, the wishgranter away mission has boxes of 10mm ammo scattered throughout it that are intended to be used with the stechkin(s) that you can find in that away mission.

**Comment by partyaddict** [comment]
> 
> 
> You do realize the stechkin is from the same era as the makarov, right? They were in service at pretty much the exact same time.

fuck

**Comment by necromanceranne** [comment]
Travis is yelling  at me right now for broken maps so gimme a bit.

**Comment by optimumtact** [comment]
hate how we have so many ammo types

**Comment by Pugie** [comment]
While we're at it make stechkin aps fit in holsters...

**Comment by necromanceranne** [comment]
Oooh good point.

**Comment by senjx** [comment]
i don't like the C-20R or the MGL change but the stetchkin and makarov changes are good but i would like to see a slight fire delay added on the makarov just to not make it a god-tier firearm out of the box

**Comment by bluetshirtguy** [comment]
fucking sweet lookin, nice.

**Comment by necromanceranne** [comment]
The benefit of the LMG is largely it's magazine. The c-20r has a magazine size of 24, the m-90gl has 30, and the lmg 50. By comparison, the APS has 15, which is about 5 shots before you empty your mag. the smg is 7, the m-90 10, and the lmg about 16?

This can make a difference in the middle of a fight. Being able to just kind of keep clicking and keep shooting as you go forward is a useful ability and worth the price. The other guns end up having to reload several times by the time the LMG has probably killed a good dozen or so people.

I'll be honest, I don't genuinely know what kind of role the m-90gl is meant to fill but innately armor piercing is interesting and does give it some value over the other guns.

**Comment by necromanceranne** [comment]
Oh and the APS should be fitting into the standard array of holsters. The only one it DOESN'T is the detective's. I could change that. It actually fits pistol ammo already.

**Comment by Tlaltecuhtli** [comment]
nerf l6 saw

**Comment by necromanceranne** [comment]
Why?

**Comment by HomocidlePancakes** [comment]
> It's honestly soothing to my OCD knowing that the soviet era guns are a little closer to their real world equivalents.

> Makarov shoots 9mm

It shoots 9x18 ROUNDS REEEEEEEEE IT’S DIFFERENT

**Comment by necromanceranne** [comment]
A LITTLE CLOSER, IT'S NOT IDENTICAL AAAAAH

**Comment by necromanceranne** [comment]
Anyway because I loved the idea of spraying the APS everywhere while dual-wielding the guns, I've increased the spread and fire rate of the gun siginificantly. Good luck hitting anything with it but it'll at least be fun. I also made the magazines in the Spetznaz bundle incendiaries to fit the theme.

I've changed 5.556mm to the old damage value but now it is innately armor piercing. The m90 also has a little bit of spread. However, it also comes with a special new ammo type. Phasic bullets. These do very small (comparatively) amounts of damage but can phase through everything.  Machine gun people down through everything and everyone! Ask relevant questions like 'why are my bullets haunted and whispering to me?' and then not get an answer! Never receive a refund even after your entire lineage is cursed by the gun ghosts! Fun for the whole family!

And finally I changed up the sprite on the APS some more and tided the code.

Also I have no idea why travis is angry!

**Comment by necromanceranne** [comment]
Wouldn't you know it, penetrator rounds and these new rounds are broken.

Aaaaaaaaaaaaaaah

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -414,6 +414,12 @@ Class Procs:
 		update_icon()
 		updateUsrDialog()
 
+/obj/machinery/CanAllowThrough(atom/movable/mover, turf/target)
+	.= ..()
```
Comment (Rohesie) [inline code/game/machinery/_machinery.dm:418]:
```suggestion
	. = ..()
```

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -414,6 +414,12 @@ Class Procs:
...
+	
+	if(istype(mover) && (mover.pass_flags & PASSMACHINE))
```
Comment (Rohesie) [inline code/game/machinery/_machinery.dm:420]:
```suggestion
	if(mover.pass_flags & PASSMACHINE)
```
Only movable atoms can move.

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSMACHINE))
```
Comment (Rohesie) [inline code/game/machinery/doors/door.dm:150]:
```suggestion
	if(mover.pass_flags & PASSMACHINE)
```

Comment (Rohesie) [reply code/game/machinery/doors/door.dm:150]: Wait, isn't this already done on the parent?

File: code/game/objects/structures.dm [hunk 4]
```diff
@@ -113,6 +113,12 @@
 		if(examine_status)
 			. += examine_status
 
+/obj/structure/CanAllowThrough(atom/movable/mover, turf/target)
+	.= ..()
+	
+	if(istype(mover) && (mover.pass_flags & PASSSTRUCTURE))
```
Comment (Rohesie) [inline code/game/objects/structures.dm:119]:
```suggestion
	. = ..()
	
	if(mover.pass_flags & PASSSTRUCTURE)
```

File: code/modules/projectiles/ammunition/ballistic/pistol.dm [hunk 5]
```diff
@@ -33,8 +33,13 @@
...
 	desc = "A 9mm armor-piercing bullet casing."
 	projectile_type =/obj/projectile/bullet/c9mm_ap
+	
+/obj/item/ammo_casing/c9mm/hp
+	name = "9mm hollow-point bullet casing"
+	desc = "A 10mm hollow-point bullet casing."
+	projectile_type = /obj/projectile/bullet/c9mm_hp
 
-/obj/item/ammo_casing/c9mm/inc
+/obj/item/ammo_casing/c9mm/fire
```
Comment (Rohesie) [inline code/modules/projectiles/ammunition/ballistic/pistol.dm:42]: `/obj/item/ammo_casing/c9mm/incendiary` might be better
`fire` is used for a lot of things on the codebase

File: code/modules/projectiles/boxes_magazines/external/pistol.dm [hunk 6]
```diff
@@ -39,17 +21,61 @@
...
+
+/obj/item/ammo_box/magazine/m9mm_aps
+	name = "stechkin pistol magazine (9mm)"
+	icon_state = "9mmaps-15"
+	ammo_type = /obj/item/ammo_casing/c9mm
+	caliber = "9mm"
+	max_ammo = 15
+
+/obj/item/ammo_box/magazine/m9mm_aps/update_icon()
+	..()
```
Comment (Rohesie) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:61]:
```suggestion
	. = ..()
```

**Comment by Rohesie** [comment]
Personally I'd prefer weapons whose design is based on old models if needed, but not outright old models.

File: code/game/machinery/doors/door.dm (hunk 3 above)
Comment (necromanceranne) [reply code/game/machinery/doors/door.dm:150]: For some reason due to the pass check already on the door, I had to add it here as well just to make sure it was checking for this flag. Not sure what a better way would be but it does work where it didn't before.

File: code/modules/projectiles/ammunition/ballistic/pistol.dm (hunk 5 above)
Comment (necromanceranne) [reply code/modules/projectiles/ammunition/ballistic/pistol.dm:42]: It's not really standardized. I could go through and do all that just to tidy it up and make it more readable

**Comment by necromanceranne** [comment]
Well we had been using the 'stechkin' for some time, and presumably if that is based on an old soviet gun, there isn't much difference between that and a makarov. There didn't seem to be any guns appearance wise similar to our stechkin designed by Stechkin that I could find. It actually looked like a makarov. And the old APS also looked like this probably because the Stechkin APS is very reminiscent in appearance to the Makarov.

What exactly our stechkin was is a mystery to me. Stechkin wasnt a brand. It was the name of its designer. Maybe it was named that because the name was obscure enough to be passed off as something obscure, old and off-market to work in with the theme of Syndicate weapons being outdated by the games standards. As an alternative, I could just restore the old sprite, rename it to something original and restore it to 10mm and retain the new APS. Though I did like the cross ammo exchangeable nature of the APS and the pistols.

**Comment by Rohesie** [comment]
Anything is good to me, it's not a hard opinion besides stating a personal preference on the guns matching the setting.

**Comment by necromanceranne** [comment]
The APS is already in the uplink next to the mosin. It would seem thr Syndicate have an assortment of old soviet weaponry for whatever reason.

**Comment by Rohesie** [comment]
:^)

File: code/game/machinery/doors/door.dm [hunk 7]
```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(mover.pass_flags & PASSMACHINE)
+		return TRUE
+
```
Comment (Rohesie) [inline code/game/machinery/doors/door.dm:152]: Isn't this already handled on the parent? Looks like a repeated check.

File: code/game/machinery/doors/door.dm [hunk 8]
```diff
@@ -146,6 +146,10 @@
 
 /obj/machinery/door/CanAllowThrough(atom/movable/mover, turf/target)
 	. = ..()
+	
+	if(mover.pass_flags & PASSMACHINE)
+		return TRUE
```
Comment (Rohesie) [inline code/game/machinery/doors/door.dm:151]:
```suggestion
	if(.)
		return
```

File: code/modules/projectiles/ammunition/ballistic/pistol.dm (hunk 5 above)
Comment (Rohesie) [reply code/modules/projectiles/ammunition/ballistic/pistol.dm:42]: That'd be a pretty positive change, though it's not really required.

File: code/game/machinery/doors/door.dm (hunk 8 above)
Comment (necromanceranne) [reply code/game/machinery/doors/door.dm:151]: oh that works? Cool.

**Comment by Rohesie** [comment]
Code is fine. Sprites pending approval by Twaticus.

**Review by Twaticus — APPROVED** [review]

**Pull request merged**